modules/local/bedtools_genomic_origin_of_reads.nf

```diff
@@ -0,0 +1,67 @@
+process BEDTOOLS_GENOMIC_ORIGIN_OF_READS {
+    tag "$meta.id"
+    label 'process_medium'
+
+    container "flomicsbiotech/flomics_qc_rnaseq:dev"
+
+    input:
+    tuple val(meta), path(bam)
+    path  gtf
+
+    output:
+    tuple val(meta), path("*.txt"), emit: results
+    tuple val(meta), path("*.tsv"), emit: table
+    path  "versions.yml"          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    shell:
+    '''
+    #take only exon records
+    awk '$3 == "exon"' !{gtf} > filtered_annotation_exon.gtf
+
+    #extract gene locus coordinates
+    cat !{gtf} | extract_locus_coords.pl - > gencode.genes.bed
+
+    #mapped fragments
+    samtools view -F4 !{bam} | cut -f1 | sort | uniq > !{meta.id}_mapped_fragments.list.txt
```

Comment:
correction: replace with `samtools view -F2304` throughout in order to remove supplementary alignments too

Reply:
All of the samtools flags have been changed to `-F2304`

Reply:
as discussed today with Silvia PL, please try with -F3332 to remove marked duplicates too, if justified

Reply:
It seems justified when running samples without UMIs. Duplicates will be removed with `umi_dedup` but duplicates on samples without UMIs are just marked. This way we make sure the module works for both types of samples. It should not make a difference on UMI samples but it will create different (and more correct) results on samples without UMIs.

Reply:
So I think in this case we should use -F3332 ?